KOGITO-8674: Update Postgresql schema migration guide for Flyway with JDBC addon (#300)
kiegroup/kogito-docs: serverlessworkflow/modules/ROOT/pages/persistence/persistence-with-postgresql.adoc

For example, when a process reaches a `callback` or needs to wait for an event, then the execution of the process is paused and the engine takes a snapshot of the workflow data. The snapshot is persisted in the database as a binary format along with process metadata information. The process metadata information includes process ID, process instance ID, and process version.

Runtime persistence is used for storing data, which is required to resume the workflow execution of a process instance. Once a process is completed, the related data is removed from the database. This means that only required data to resume the execution is persisted.

In {product_name}, you can enable persistence using add-ons. This document describes the use of the `kogito-addons-quarkus-persistence-jdbc` add-on, which is based on Java Database Connectivity (JDBC) along with PostgreSQL.

The `kogito-addons-quarkus-persistence-jdbc` add-on also extends on the Quarkus capabilities and you can use the available features directly from Quarkus JDBC support. For more information about Quarkus and JDBC, see link:{quarkus_datasource_guide}[Quarkus Datasources].

> You can also see the `serverless-workflow-callback-quarkus` example application in GitHub repository. To execute the `serverless-workflow-callback-quarkus` example application, you can follow the instructions mentioned in the `README` file. To clone the `kogito-example` repository, use the following command:

Reviewer: maybe link the example directly here, wdyt?

. Create PostgreSQL database schema.
+
--

> The persistence add-on uses link:{flyway_url}[Flyway] to initialize database schema. The add-on does not create the required database schema structure by default. To enable the creation of database schema by default, you must set the following properties in `application.properties` file.

Reviewer 2: ```suggestion
The persistence add-on uses link:{flyway_url}[Flyway] to initialize database schema. The add-on does not create any required database schema structure by default. To enable the creation of the database schema by default, you must set the following properties in the `application.properties` file.
```

> The persistence add-on uses link:{flyway_url}[Flyway] to initialize database schema. The add-on does not create any required database schema structure by default. To enable the creation of the database schema by default, you must set the following properties in the `application.properties` file.

Reviewer 3: ```suggestion
The persistence add-on uses link:{flyway_url}[Flyway] to create the database schema.  To enable the database schema creation during the runtime startup, you must set the following properties in the `application.properties` file.
```

[source,properties]
----
quarkus.flyway.migrate-at-start=true
quarkus.datasource.db-kind=postgresql
----

You can find more details regarding the PostgreSQL database schema migration in the xref:persistence/postgresql-flyway-migration.adoc[Flyway migration guide].
--

> . Optional: To handle the concurrent requests to shared workflow instances, enable the persistence-enabled optimistic locking for concurrency control using the version field in the database.

Reviewer: for shared?

  "version": "1.0"
}
----
The versioning strategy is used to allow different workflow applications to run different versions of a process at the same time. The different versions of a process share the same database. This is useful when you migrate a process from one version to another. When allowing workflow instances to finish executing, a new version can be deployed using a new workflow application setup.

By default, the engine considers the `version` specified in the workflow file as the current version of the asset. Therefore, you need to manually change the `version` in the workflow file, making the engine consider the specified version as a new version.

> As an alternative, you can set the `kogito.workflow.version-strategy=project` property in the `application.properties` file of your project. This enables the engine to consider the Maven or Gradle project version as the version of all workflows in the project. For instance, when you release a new version of your Maven project, the version in the workflow file is automatically updated.

Reviewer: besides project strategy, is there any other?

Reviewer 3: unrelated changes to the JIRA, @akumar074 maybe review it so it doesnt create confusion regarding the content.
